src/avt/Database/Ghost/avtStructuredDomainBoundaries.C

```diff
@@ -1815,16 +1818,23 @@ avtStructuredDomainBoundaries::ExchangeScalar(vector<int>           domainNum,
 {
     int dataType = (scalars.empty() ? -1 : scalars[0]->GetDataType());
 
+    int maxDataType = dataType;
 #ifdef PARALLEL
     // Let's get them all to agree on one data type.
-    int myDataType = dataType;
-    MPI_Allreduce(&myDataType, &dataType, 1, MPI_INT, MPI_MAX, VISIT_MPI_COMM);
+    MPI_Allreduce(&dataType, &maxDataType, 1, MPI_INT, MPI_MAX, VISIT_MPI_COMM);
 #endif
-
-    if (dataType < 0)
+    if (maxDataType < 0)
         return scalars;
 
-    switch (dataType)
+    if((dataType >= 0) && (dataType != maxDataType))
+    {
+        // This should never happen, so throw the exception.
+        EXCEPTION1(VisItException,
```

Comment:
@markcmiller86 I figured out that an Exception here isn't good unless all procs call the exception (it causes a hang).  What would you suggest?

Reply:
I figured that  would be be case. So, I think just return; as you were before. Not optimal but way better than crashing or deadlocking.

Reply:
that still won't work, because one process will try the Exchange but the other will exit early.
Write to debug log but let the function continue?

Reply:
Doh! Of course, that second piece (`maxDataType != dataType`) will be known only on the calling processor.

It needs to be a reduce of both the min and the max. I thought MPI had a built in MINMAX reduce. I don't see one. I do see on in `avtParallelContext` (for doubles).

```

double minmaxDataType[2] = {(double) dataType, (double) dataType};
UnifyMinMax(minmaxDataType, 2, 0);

if (minmaxDataType[0] < 0 || minmaxDataType[0] != minmaxDataType[1])
```

Reply:
Would using UnifyMinMax be better than the change I just made?

Reply:
Never mind, the new way is cleaner, implementing now.

Reply:
Actually, using the UnifyMinMax doesn't work when one proc doesn't have data, because its dataType is set to -1.
My current method is ugly, but it works.

Reply:
Ok, I am obviously totally misunderstanding and have lead you on a wild goose-chase. Sorry for trobules :sad:

Reply:
BTW...thanks very much for your efforts, nonetheless 💪🏻

Reply:
FWIW, I can test that the Exception is thrown and doesn't cause a hang by changing 
```
int hasDataTypeMismatch = ((dataType >= 0) && (dataType != maxDataType));
```
to

```
int hasDataTypeMismatch =(dataType != maxDataType);
```

Then run the silo.py test, silo_45 triggers the exception, there will be an exception message printed, but the regressions move on.

Reply:
hopefully its never triggered IRL.